_utils/_translation_utils/post_transifex_pull.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+# to be run from the git root
+# $1 is lang
+# $2 is directory where translated files reside and language needs to be added to internal urls
+# TODO param check
+
+set -e
+
+echo "============================ post processing step 1 ======================================"
+#read b
+bash _utils/_translation_utils/prepare_tx_config_postprocess.sh .tx/config /tmp/tx-mapping
+
+
+echo "============================ post processing step 2 ======================================"
+#read b
+ruby _utils/_translation_utils/merge_md_heading_ids.rb "$1" /tmp/tx-mapping
```

Comment:
At least for the `merge_md_heading_ids.rb` script, called here, it is important that the checked out commit is not the most recent one but that commit that was the current one when the files have been pushed to Transifex the last time. The reason is that headlines may have changed or added or removed meanwhile and thus, the `merge_md_heading_ids.rb` script may produce a wrong output.

I can't see any attempt to checkout that correct commit or have I overseen something?

IMHO when pushing files to Transifex, the related commit hash should be saved. When pulling the files from Transifex, that related (old) commit should be checked out for the postprocessing steps.

Is it understandable what I mean?

Reply:
This all is called just after _pushing_ and pulling from transifex. See https://github.com/QubesOS/qubesos.github.io/pull/154/files#diff-037ea159eb0a7cb0ac23b851e66bee30fb838ee8d0d99fa331a1ba65283d37f7R38-R39

Reply:
Ah, okay, now I see. Both pushing and pulling belong to the same commit and thus everything is right. Thank you for clarification!